sagemaker-python-sdk/tensorflow_script_mode_mnist/tensorflow_distributed_mnist.ipynb

```diff
@@ -0,0 +1,224 @@
+    "\n",
+    "The training script was adapted from TensorFlow's official [CNN MNIST example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/tutorials/layers/cnn_mnist.py). We have modified to hanle a ``model_dir`` parameter passed in by SageMaker. This is a S3 path which can be used for data sharing during distributed training and checkpoint and or model persistence. In addtion we have also added a argument parsing function to handle processing training related variables.\n",
+    "\n",
+    "At the end of the training job we also added a step to export the trained model to the path stored in the environment variable ``SM_MODEL_DIR`` which always points to ``/opt/ml/model``. This is critical because SageMaker uploads all the model artifact in this folder to S3 at end of training.\n",
+    "\n",
+    "Here is the entire script:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "!cat 'mnist.py'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a training job using the sagemaker.TensorFlow estimator\n",
+    "\n",
+    "The ``sagemaker.TensorFlow`` estimator handles locating the script mode container uploading your script to a S3 location and creating a SageMaker training job."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "from sagemaker.tensorflow import TensorFlow\n",
+    "\n",
+    "\n",
+    "mnist_estimator = TensorFlow(entry_point='mnist.py',\n",
+    "                             role='SageMakerRole',\n",
+    "                             train_instance_count=2,\n",
+    "                             train_instance_type='ml.p2.xlarge',\n",
+    "                             framework_version='1.12',\n",
+    "                             py_version = 'py3',\n",
+    "                             distributions = {'parameter_server': {'enabled': True}},\n",
+    "                             base_job_name='test-tf')\n",
+    "\n",
+    "mnist_estimator.fit(training_data_uri)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The **```fit```** method will create a training job in two **ml.p2.xlarge** instances. The logs above will show the instances doing training, evaluation, and incrementing the number of **training steps**. \n",
+    "\n",
+    "In the end of the training, the training job will generate a saved model for TF serving."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "# Deploy the trained model to prepare for predictions\n",
+    "\n",
+    "The deploy() method creates an SageMaker model which is then used to create an endpoint which serves prediction requests in realtime. "
```

Comment:
Same for deploy here. I would mentioned more about the TFS container and what is necessary to deploy a TFS model.